dpm/client/__init__.py

```diff
@@ -9,21 +9,30 @@
 class DpmException(Exception):
     pass
 
+
+class ConfigError(Exception):
+    """ The configuration passed to the client is malformed """
+    pass
+
+
 class Client(object):
 
-    def __init__(self, data_package_path='', config_path="~/.dpm/config", click=None):
+    def __init__(self, config, data_package_path='', click=None):
         if not data_package_path:
             data_package_path = os.getcwd()
         data_package_path = os.path.abspath(data_package_path)
         # may want to use the datapackage-py here
         self.datapackage = self._load_dp(data_package_path)
 
-        # load config ... 
-        self.config = self._load_config(config_path)
+        try:
```

Comment:
Generally i think we just want in the `__init__`:

self.config = config

Then somewhere where we need config info - e.g. in publish or other methods we can use a `self._ensure_config` or whatever (probably in ensure_auth etc).